packages/ipfs-http-client/src/index.js

```diff
@@ -8,14 +8,114 @@ const multicodec = require('multicodec')
 const multihash = require('multihashes')
 const globSource = require('ipfs-utils/src/files/glob-source')
 const urlSource = require('ipfs-utils/src/files/url-source')
+const toUri = require('multiaddr-to-uri')
+const { isBrowser, isWebWorker } = require('ipfs-utils/src/env')
+const { URL } = require('iso-url')
+const API = require('./lib/api')
+
+const isMultiaddr = (input) => {
+  try {
+    multiaddr(input) // eslint-disable-line no-new
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+const normalizeURL = (config) => {
+  let api
+  if (typeof config === 'string') {
+    api = config
+  }
+
+  if (multiaddr.isMultiaddr(config) || isMultiaddr(config)) {
+    api = toUri(config)
+  }
+
+  const url = new URL(api)
+
+  url.pathname = config.apiPath || 'api/v0'
+  if (!api) {
+    if (isBrowser || isWebWorker) {
+      url.protocol = config.protocol || location.protocol
+      url.host = config.host || location.hostname
+      url.port = config.port || location.port
+    } else {
+      url.host = config.host || 'localhost'
+      url.port = config.port || '5001'
+      url.protocol = config.protocol || 'http'
+    }
+  }
+  return url
+}
+
+const errorHandler = async (response) => {
+  let msg
+
+  try {
+    if ((response.headers.get('Content-Type') || '').startsWith('application/json')) {
+      const data = await response.json()
+      // log(data)
+      msg = data.Message || data.message
+    } else {
+      msg = await response.text()
+    }
+  } catch (err) {
+    // log('Failed to parse error response', err)
+    // Failed to extract/parse error message from response
+    msg = err.message
+  }
+
+  const error = new API.HTTPError(response)
+
+  // If we managed to extract a message from the response, use it
+  if (msg) {
+    error.message = msg
+  }
+
+  throw error
+}
+
+var KEBAB_REGEX = /[A-Z\u00C0-\u00D6\u00D8-\u00DE]/g
```

Comment:
Might be helpful to insert a comment above this line explaining in plain text what this regex is intended to catch beyond A-Z

Reply:
@hugomrdias Curious for your thoughts on this when you get a chance. Thanks!